libs/server-sdk/src/data_interfaces/source/ipush_source.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <launchdarkly/data_model/sdk_data_set.hpp>
+
+#include <functional>
+#include <optional>
+#include <string>
+
+namespace launchdarkly::server_side::data_interfaces {
+
+/**
+ * \brief IPushSource obtains data via a push synchronization mechanism,
```

Comment:
I think we need a more elaborate explanation on this one, because the name and the content don't make sense to me.

Maybe the actual pushing happens outside the interface, which is confusing.